src/editor/Store/NeovimStoreConnector.re

```diff
@@ -88,30 +90,82 @@ let start = (executingDirectory, setup: Core.Setup.t, cli: Core.Cli.t) => {
       }
     );
 
-  let updater = (state, action) => {
+  let detachBufferEffect = id =>
+    Isolinear.Effect.create(~name="neovim.detachBuffer", () =>
+      neovimProtocol.bufDetach(id)
+    );
+
+  let shouldAttachUIEffect = ({neovimActive, _}: Model.State.t) =>
+    neovimActive ? Isolinear.Effect.none : attachUIEffect;
+
+  let updater = (state: Model.State.t, action) =>
     switch (action) {
+    | Model.Actions.OpenHome =>
+      let id = state.activeBufferId;
+      (state, detachBufferEffect(id));
+    | Model.Actions.StartEditor =>
+      /**
+      We attach to neovim as a separate action rather than immediately
```

Comment:
This would mean that you can't open the command line in the welcome screen, right? So I couldn't run `:new` from the welcome screen?

Reply:
not necessarily the only thing about this approach is that it requires us to be explicit about attaching to neovim rather than implicitly always being attached. I tried quite a few different approaches and this was the only one that I found that would allow us to have a gui tab independent from neovim without creating a fake neovim tab etc.